### PR TITLE
model: ImageTarget has both localRef and clusterRef [ch4292]

### PR DESCRIPTION
**Pull request by maiamcc**
not ever populating these refs with different things yet; that behavior
and tests to follow

### CODE REVIEW

File: pkg/model/image_target.go [hunk 1]
```diff
@@ -4,17 +4,14 @@ import (
...
 	"github.com/windmilleng/tilt/internal/container"
 	"github.com/windmilleng/tilt/internal/sliceutils"
 )
 
 type ImageTarget struct {
-	ConfigurationRef container.RefSelector
-	DeploymentRef    reference.Named
-	BuildDetails     BuildDetails
-	MatchInEnvVars   bool
+	Refs           container.RefSet
```
Comment (nicks) [inline pkg/model/image_target.go]: out of curiosity, why group them together into a sub-struct? it doesn't seem like you ever use them together as a group

Comment (nicks) [reply pkg/model/image_target.go]: it might make sense to add ConfigurationRef() etc accessors

File: pkg/model/manifest_test.go [hunk 2]
```diff
@@ -141,29 +141,43 @@ var equalitytests = []struct {
...
+		Manifest{}.WithImageTarget(ImageTarget{Refs: container.RefSet{LocalRef: img1.AsNamedOnly()}}),
+		Manifest{}.WithImageTarget(ImageTarget{Refs: container.RefSet{LocalRef: img2.AsNamedOnly()}}),
 		false,
 		true,
 	},
 	{
-		"ImageTarget.DeploymentRef equal",
-		Manifest{}.WithImageTarget(ImageTarget{DeploymentRef: img1.AsNamedOnly()}),
-		Manifest{}.WithImageTarget(ImageTarget{DeploymentRef: img1.AsNamedOnly()}),
+		"ImageTarget.LocalRef equal",
```
Comment (nicks) [inline pkg/model/manifest_test.go]: fwiw, it doesn't seem like adding tests here is adding a ton of value. Now that we've added the comparator library, they're basically just testing the comparator library.

File: pkg/model/image_target.go (hunk 1 above)
Comment (maiamcc) [reply pkg/model/image_target.go]: in a future PR i pass refsets around to builders -- i don't feel strongly about it, it just seemed a way to reduce the number of args to keep track of 🤷‍♀

File: pkg/model/manifest_test.go (hunk 2 above)
Comment (maiamcc) [reply pkg/model/manifest_test.go]: the what now?

**Pull request merged**